addons/event/migrations/11.0.1.0/pre-migration.py

```diff
@@ -0,0 +1,9 @@
+# Copyright 2018 Tecnativa - Vicent Cubells
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    env.ref('event.report_event_registration_company_rule').unlink()
```

Comment:
You have to prevent if the rule has been manually deleted or there's a constraint:
```python
try:
    env.ref('event.report_event_registration_company_rule').unlink()
except Exception:
    pass
```